solarforecastarbiter/io/reference_observations/pnnl.py

```diff
@@ -0,0 +1,168 @@
+
+import logging
+from pathlib import Path
+
+import pandas as pd
+
+from solarforecastarbiter.io.reference_observations import (
+    common, default_forecasts)
+
+logger = logging.getLogger('reference_data')
+
+DATA_PATH = Path('pnnl_data')
+
+# These columns are just the minute data. They do not include the daily
+# summary columns.
+COLUMNS = [
+    'ID',
+    'YEAR',
+    'DAY',
+    'HRMN',
+    'PIR_avg(W/m2)',
+    '848_avg(W/m2)',
+    'NIP_avg(W/m2)',
+    'PSP_avg(W/m2)',
+    'IRnet_avg(W/m2)',
+    'PIR_std(W/m2)',
+    '848_std(W/m2)',
+    'NIP_std(W/m2)',
+    'PSP_std(W/m2)',
+    'Tdome_avg(C)',
+    'Tcase_avg(C)',
+    'Tair_avg(C)',
+    'RHair_avg(%)',
+    'T10X_avg(C)',
+    'p10X_avg(V)',
+]
+
+# variables to insert into database
+VARIABLE_MAP = {
+    'PSP_avg(W/m2)': 'ghi',
+    'NIP_avg(W/m2)': 'dni',
+    '848_avg(W/m2)': 'dhi',
+    'Tair_avg(C)': 'air_temperature',
+    'RHair_avg(%)': 'relative_humidity'
+}
+
+
+def initialize_site_observations(api, site):
+    """Creates an observation at the site for each VARIABLE_MAP.values().
+
+    Parameters
+    ----------
+    site : datamodel.Site
+        The site object for which to create Observations.
+    """
+    try:
+        extra_params = common.decode_extra_parameters(site)
+    except ValueError:
+        logger.warning('Cannot create reference observations at PNNL site '
+                       f'{site.name}, missing required parameters.')
+        return
+    for pnnl_var, sfa_var in VARIABLE_MAP.items():
+        obs_extra_params = extra_params.copy()
+        obs_extra_params['network_data_label'] = pnnl_var
+        logger.info(f'Creating {sfa_var} at PNNL')
+        common.create_observation(
+            api, site, sfa_var, extra_params=obs_extra_params
+        )
+
+
+def initialize_site_forecasts(api, site):
+    """
+    Create forecasts for each variable in VARIABLE_MAP.values().
+
+    Parameters
+    ----------
+    api : solarforecastarbiter.io.api.APISession
+        An active Reference user session.
+    site : datamodel.Site
+        The site object for which to create Forecasts.
+    """
+    common.create_forecasts(api, site, VARIABLE_MAP.values(),
+                            default_forecasts.TEMPLATE_FORECASTS)
+
+
+def fetch(api, site, start, end):
+    """Retrieve observation data for PNNL site for 1 year.
+
+    Assumes that data is available in a directory structure
+    ``pnnl_data/rldradC1.00.{%Y}/rldradC1.00.%Y%m%d.%H%M%S.raw.%Y%m%D%Hpnl.sky``
+
+    Parameters
+    ----------
+    api : io.APISession
+        Unused but conforms to common.update_site_observations call
+    site : datamodel.Site
+        Unused but conforms to common.update_site_observations call
+    start : datetime
+        The beginning of the period to request data for.
+    end : datetime
+        The end of the period to request data for. If ``end.year >
+        start.year``, data will only be pulled through ``start.year``.
```

Comment:
Why? seems not so hard to loop through multiple years. I understand one might not be able to post that much at once